src/windows/win-crowdstrike-fix-bootloop-v2.ps1

```diff
@@ -1,11 +1,124 @@
 . .\src\windows\common\setup\init.ps1
 . .\src\windows\common\helpers\Get-Disk-Partitions-v2.ps1
 
-$partitionlist = Get-Disk-Partitions
-$driveLetters = $partitionlist.DriveLetter
-Log-Info "Found drive letters: $driveLetters"
-$actionTaken = $false
 
+# Check if corrupt CrowdStrike files exist in each drive letter
+# Remove any corrupt CrowdStrike files
+function RemoveCrowdStrikeFiles
+{
+    param(
+    [Parameter(Mandatory = $true)]
+        [Object[]]$Partitionlist
+    )
+
+    Log-Info "Removing any corrupt crowdstrike files..."
+    $crowdStrikeFileRemoved = $false
+    forEach ( $partition in $partitionlist )
+    {
+        $driveLetter = $partition.DriveLetter
+        if ($driveLetter) { # Skip partitions without drive letter
+            $driveLetter = ($driveLetter + ":")
+            Log-Info "Check Drive letter: $driveLetter"
+            $corruptFiles = "$driveLetter\Windows\System32\drivers\CrowdStrike\C-00000291*.sys"
+
+            if (Test-Path -Path $corruptFiles) {
+                Log-Info "Found crowdstrike files to cleanup at $corruptFiles, removing..."
+                Remove-Item $corruptFiles
+                Log-Info "Corrupt crowdstrike files are removed."
+                $crowdStrikeFileRemoved = $true
+            }
+        }
+    }
+
+    if ($crowdStrikeFileRemoved) {
+        Log-Info "Successfully cleaned up crowdstrike files"
+    } else {
+        Log-Warning "No bad crowdstrike files found"
+    }
+}
+
+# Check if registry config files exist in each non system drive letter.
+# if registry config files exist in the non system drive letter, load it to the registry hive and then unload it.
+function LoadUnloadRegistryHives
+{
+    param(
+    [Parameter(Mandatory = $true)]
+        [Object[]]$Partitionlist
+    )
+    Log-Info "Loading/unloading Registry Hives from registry config files..."
+
+    # System Drive (which is usually C:) should be skipped as it is from the OS disk rather than the Data disk
+    Log-Info "Getting system drive..."
+    $systemDrive = $Env:SYSTEMDRIVE
+    Log-Info "System drive is: $systemDrive"
+
+    $registryConfigFileFound = $false
+    forEach ( $partition in $partitionlist )
+    {
+        $driveLetter = $partition.DriveLetter
+        if ($driveLetter) { # Skip partitions without drive letter
+            $driveLetter = ($driveLetter + ":")
+            Log-Info "Check Drive letter: $driveLetter"
+            if ($driveLetter -ne $systemDrive) { # Skip OS disk
```

Comment:
can we make this case insensitive check?

Reply:
The "Not Equal" comparison here is not case sensitive:
![image](https://github.com/user-attachments/assets/b646a893-70c2-4c55-a68d-65bade6fcb12)

We should not do case sensitive comparation as "C:" drive and "c:" drive is the same drive in Windows.

Reply:
I agree and was saying case insensitive, which looks like -ne does by default. So, that's good.